modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/TracingLevelExportInfo.java

```diff
@@ -9,64 +9,100 @@
+                                      !tracingLevelOverride.equals(tracingLevel));
   }
 
-  public boolean isOverride() {
-    return this.isOverride;
+  public static TracingLevelExportInfo createTracingLevelExportInfo(Component component, String nameOverride,
+                                                                    TracingLevelConfiguration tracingLevelConfiguration) {
+    // Porque se pide un override si puede ser igual al level? Agregar un isOverride a TracingLevel
+    TracingLevel tracingLevel = tracingLevelConfiguration.getTracingLevel();
+    TracingLevel tracingLevelOverride =
+        getTracingLevelOverride(component, tracingLevelConfiguration);
+    return new TracingLevelExportInfo(resolveInitialExportInfoProvider(tracingLevelOverride), nameOverride,
+                                      !tracingLevelOverride.equals(tracingLevel));
   }
 
-  public boolean isExportable() {
-    if (initialExportInfo == null) {
-      initialExportInfo = getInitialExportInfo();
-    }
-    return initialExportInfo.isExportable();
+  private static TracingLevel getTracingLevelOverride(Component component, TracingLevelConfiguration tracingLevelConfiguration) {
+    return component.getLocation() != null
+        ? tracingLevelConfiguration.getTracingLevelOverride(component.getLocation().getLocation())
+        : tracingLevelConfiguration.getTracingLevel();
   }
 
-  public Set<String> noExportUntil() {
-    if (initialExportInfo == null) {
-      initialExportInfo = getInitialExportInfo();
+  private static InitialExportInfoProvider resolveInitialExportInfoProvider(TracingLevel tracingLevel) {
+    switch (tracingLevel) {
+      case OVERVIEW:
+        return new OverviewInitialExportInfoProvider();
+      case DEBUG:
+        return new DebugInitialExportInfoProvider();
+      default:
+        return new MonitoringInitialExportInfoProvider();
     }
-    return initialExportInfo.noExportUntil();
   }
 
+  // Esto tendria que ser mas estatico (revisar)
   private InitialExportInfo getInitialExportInfo() {
-    if (this.spanIdentifier instanceof Component) {
-      return this.initialExportInfoProvider.getInitialExportInfo((Component) spanIdentifier);
+    if (spanNameOverride != null) {
+      return initialExportInfoProvider.getInitialExportInfo(spanNameOverride);
+    } else {
+      return initialExportInfoProvider.getInitialExportInfo(component);
     }
-    return this.initialExportInfoProvider.getInitialExportInfo((String) spanIdentifier);
   }
 
-  public void propagateExportInfo(TracingLevelExportInfo parentTracingLevelExportInfo) {
-    if (!isOverride() && parentTracingLevelExportInfo.isOverride()) {
-      this.initialExportInfoProvider = parentTracingLevelExportInfo.getInitialExportInfoProvider();
-      initialExportInfo = getInitialExportInfo();
-      this.isOverride = true;
-    }
+  public boolean isExportable() {
+    return getInitialExportInfo().isExportable();
```

Comment:
Is there any way we can still avoid calling getInitialExportInfo() every time isExportable() and noExportUntil() are called? Because of performance improvements on both these methods the following if was added before on this [PR](https://github.com/mulesoft/mule/commit/6e1a2ed53d482a4c0fce3085ead6f59b89a670e1):
```
if (initialExportInfo == null) {
      initialExportInfo = getInitialExportInfo();
}  
```
Can this be kept somehow?

Reply:
I think thats being called only once so you will not see any improvement by adding that check, but if you look at the getInitialExportInfo method it has a comment about making it more static so yes, we can either add that check or make it a lazy value.